DirectProgramming/DPC++FPGA/Tutorials/Features/prototype/CSR_and_streaming_interfaces/README.md

```diff
@@ -0,0 +1,300 @@
+
+***Documentation***:  The [DPC++ FPGA Code Samples Guide](https://software.intel.com/content/www/us/en/develop/articles/explore-dpcpp-through-intel-fpga-code-samples.html) helps you to navigate the samples and build your knowledge of DPC++ for FPGA. <br>
+The [oneAPI DPC++ FPGA Optimization Guide](https://software.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide) is the reference manual for targeting FPGAs through DPC++. <br>
+The [oneAPI Programming Guide](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/) is a general resource for target-independent DPC++ programming.
+
+| Optimized for                     | Description
+---                                 |---
+| OS                                | Linux* Ubuntu* 18.04/20.04, RHEL*/CentOS* 8, SUSE* 15; Windows* 10
+| Hardware                          | Intel® FPGA Programmable Acceleration Card (PAC) D5005 (with Intel Stratix® 10 SX) <br> Intel® FPGA 3rd party / custom platforms with oneAPI support (and SYCL USM support) <br> *__Note__: Intel® FPGA PAC hardware is only compatible with Ubuntu 18.04*
+| Software                          | Intel® oneAPI DPC++ Compiler <br> Intel® FPGA Add-On for oneAPI Base Toolkit
+| What you will learn               | Basics of specifying kernel control interfaces and kernel argument interfaces
+| Time to complete                  | 30 minutes
+
+
+
+## Purpose
+
+Use CSR and streaming interface controls to specify how the kernel control handshaking is performed, as well as how the kernel argument data is passed in to the kernel.
+
+### Declaring a CSR Agent kernel interface
+
+#### Example Functor
+
+```c++
+struct MyIP {
+  MyIP() {}
+  register_map_interface void operator()() const {
+    ...
+  }
+};
+```
+
+#### Example Lambda
+```c++
+q.single_task([=] register_map_interface {
+  ...
+})
+```
+
+### Declaring a streaming kernel interface
+
+#### Example Functor
+
+```c++
+struct MyIP {
+  MyIP() {}
+  streaming_interface void operator()() const {
+    ...
+  }
+};
+```
+
+#### Example Lambda
+```c++
+q.single_task([=] streaming_interface {
+  ...
+})
+```
+
+### Declaring a CSR Agent kernel argument interface
+
+#### Example Functor
+
+```c++
+struct MyIP {
+  register_map int arg1;
+  MyIP(int arg1_) : arg1(arg1_) {}
+  register_map_interface void operator()() const {
+    ...
+  }
+};
+```
+
+*__Note__:* CSR agent kernel arguments are not currently supported in kernels with streaming control.
+
+### Declaring a streaming kernel argument interface
+
+```c++
+struct MyIP {
+  conduit int arg1;
```

Comment:
Conduit VS streaming 
Why do we have different names in the code and in the text?